Dungeons/src/main/java/dungeons/Player.java

```diff
@@ -0,0 +1,46 @@
+package dungeons;
+
+import static dungeons.utils.Constants.*;
+
+public class Player {
+    private int health = MAX_HP;
+    private int gold = 0;
+    private boolean isPoisoned = false;
+    private boolean isCursed = false;
+
+    public void addGold(int amount) {
+        gold += amount;
+    }
+
+    public void takeDamage(int damage) {
+        health -= damage;
+        if (health <= 0) {
+            System.out.println("You have died.");
+            System.exit(0);
```

Comment:
why does the player decide when the game ends?

Reply:
moved to game.java